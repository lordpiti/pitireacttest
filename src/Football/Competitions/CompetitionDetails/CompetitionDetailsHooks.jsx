import React, { Component, useEffect, useState } from 'react';
import { Route, Redirect } from 'react-router-dom';
import CompetitionInfo from '../CompetitionInfo/CompetitionInfo';
import CompetitionRounds from '../CompetitionRounds/CompetitionRounds';
import CompetitionDraw from '../CompetitionDraw/CompetitionDraw';
import SideMenu from '../../components/SideMenu/SideMenu';
import Match from '../Match/Match';
import CompetitionStatistics from '../CompetitionStatistics/CompetitionStatistics';
import { connect } from 'react-redux';
import * as actionCreators from '../../store/actions/competitions';

const CompetitionDetails = (props) => {

    const competitionId = props.match.params.id;
    useEffect(() => {
        props.loadCompetition(competitionId);
    },  []);
    

    let menuItemList = [
        {
            name: 'Summary',
            url: props.match.url + '/overview'
        }
    ];

    let competitionTypeContent = null;

    let pageContent = null;

    if (props.currentCompetition) {
        if (props.currentCompetition.type !== 'Playoff') {
            menuItemList = menuItemList.concat([{
                name: 'Rounds',
                url: props.match.url + '/competition-rounds'
            },
            {
                name: 'Statistics',
                url: props.match.url + '/competition-statistics'
            }]);
            competitionTypeContent = <CompetitionRounds competitionData={props.currentCompetition} {...props} ></CompetitionRounds>;
        }
        else {
            menuItemList.push({
                name: 'Draw',
                url: props.match.url + '/competition-rounds'
            });
            competitionTypeContent = <CompetitionDraw match={props.match} competitionData={props.currentCompetition} ></CompetitionDraw>;
        }
        // By putting "component=xxx" in the Route to render a component, we force that everytime 'xxx' changes,
        // the child component runs the whole life cycle, including the constructor
        // Normally this is not needed so better to use "render=xxx" which will only run the render method in the child component  
        pageContent =
            <div className="CompetitionDetails">
                <div className="row">
                    <div className="col-sm-3">
                        <SideMenu itemList={menuItemList} >
                            <div className="margin-bottom-medium">
                                <img src={props.currentCompetition.logo.url} className="roundedImage" height="50" width="50" />
                                <span>{`${props.currentCompetition.name}`}</span>
                            </div>
                        </SideMenu>
                    </div>
                    <div className="col-sm-9">
                        <Route path={props.match.url + '/'} exact
                            render={() => (<Redirect to={props.match.url + '/overview'} />)}
                        />
                        <Route path={props.match.url + '/competition-rounds/match/:id'}
                            component={Match}
                        />
                        <Route path={props.match.url + '/overview'}
                            render={() => {
                                return (
                                    <CompetitionInfo competitionData={props.currentCompetition}></CompetitionInfo>
                                    )
                            }
                            } />
                        <Route path={props.match.url + '/competition-rounds'}
                            component={() => competitionTypeContent} exact />
                        <Route path={props.match.url + '/competition-statistics'}
                            render={() => <CompetitionStatistics competitionId={competitionId}></CompetitionStatistics>} exact />
                    </div>
                </div>
            </div>
    }

    return (
        <div>
            {pageContent}
        </div>
    );
}


const mapStateToProps = state => {
    return {
        currentCompetition: state.competitions.currentCompetition
    }
};

const mapDispatchToProps = dispatch => {
    return {
        loadCompetition: (competitionId) => dispatch(actionCreators.loadCompetition(competitionId))
    }
};

export default connect(mapStateToProps, mapDispatchToProps)(CompetitionDetails);