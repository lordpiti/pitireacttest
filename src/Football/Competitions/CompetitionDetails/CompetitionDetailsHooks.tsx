import React, { useEffect } from 'react';
import { Route, Redirect } from 'react-router-dom';
import CompetitionInfo from '../CompetitionInfo/CompetitionInfo';
import CompetitionRounds from '../CompetitionRounds/CompetitionRounds';
import CompetitionDraw from '../CompetitionDraw/CompetitionDraw';
import SideMenu, { MenuItemSideMenu } from '../../components/SideMenu/SideMenu';
import Match from '../Match/Match';
import CompetitionStatistics from '../CompetitionStatistics/CompetitionStatistics';
import { useSelector } from 'react-redux';
import * as actionCreators from '../../store/actions/competitionsActions';
import { RouteComponentProps } from 'react-router';
import { FootballState, useFootballDispatch } from '../../store';

interface MatchParams {
  id: string;
}

const CompetitionDetails = (props: RouteComponentProps<MatchParams>) => {
  const competitionId = parseInt(props.match.params.id);

  const dispatch = useFootballDispatch();

  const theState = useSelector((state: FootballState) => ({
    currentCompetition: state.competitions.currentCompetition,
  }));

  useEffect(() => {
    dispatch(actionCreators.loadCompetition(competitionId));
  }, []);

  let menuItemList: MenuItemSideMenu[] = [
    {
      name: 'Summary',
      url: props.match.url + '/overview',
    },
  ];

  let competitionTypeContent = <></>;

  if (theState.currentCompetition) {
    if (theState.currentCompetition.type !== 'Playoff') {
      menuItemList = menuItemList.concat([
        {
          name: 'Rounds',
          url: props.match.url + '/competition-rounds',
        },
        {
          name: 'Statistics',
          url: props.match.url + '/competition-statistics',
        },
      ]);

      competitionTypeContent = (
        <CompetitionRounds
          competitionData={theState.currentCompetition}
        ></CompetitionRounds>
      );
    } else {
      menuItemList.push({
        name: 'Draw',
        url: props.match.url + '/competition-rounds',
      });
      competitionTypeContent = (
        <CompetitionDraw
          match={props.match}
          competitionData={theState.currentCompetition}
        ></CompetitionDraw>
      );
    }
    // By putting "component=xxx" in the Route to render a component, we force that everytime 'xxx' changes,
    // the child component runs the whole life cycle, including the constructor
    // Normally this is not needed so better to use "render=xxx" which will only run the render method in the child component
    return (
      <div className='CompetitionDetails'>
        <div className='row'>
          <div className='col-sm-3'>
            <SideMenu itemList={menuItemList}>
              <div className='margin-bottom-medium'>
                <img
                  src={theState.currentCompetition.logo.url}
                  className='roundedImage'
                  height='50'
                  width='50'
                />
                <span>{theState.currentCompetition.name}</span>
              </div>
            </SideMenu>
          </div>
          <div className='col-sm-9'>
            <Route
              path={props.match.url + '/'}
              exact
              render={() => <Redirect to={props.match.url + '/overview'} />}
            />
            <Route
              path={props.match.url + '/competition-rounds/match/:id'}
              component={Match}
            />
            <Route
              path={props.match.url + '/overview'}
              render={() => {
                return (
                  <CompetitionInfo
                    competitionData={theState.currentCompetition}
                  ></CompetitionInfo>
                );
              }}
            />
            <Route
              path={props.match.url + '/competition-rounds'}
              render={() => competitionTypeContent}
              exact
            />
            <Route
              path={props.match.url + '/competition-statistics'}
              render={() => (
                <CompetitionStatistics
                  competitionId={competitionId}
                ></CompetitionStatistics>
              )}
              exact
            />
          </div>
        </div>
      </div>
    );
  }

  return <></>;
};

export default CompetitionDetails;
