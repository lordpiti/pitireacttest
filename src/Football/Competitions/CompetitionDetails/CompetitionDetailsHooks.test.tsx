import React from "react";
import CompetitionDetailsHooks from "./CompetitionDetailsHooks";
import { shallow, mount } from "enzyme";
import { MemoryRouter } from "react-router";

jest.mock("react-redux", () => ({
  useDispatch: () => {},
  useSelector: () => ({
    currentCompetition: {
      logo: {
        url: "super",
      },
    },
  }),
  connect: () => () => {},
}));

describe("<SDGDG /> with all data", () => {
  it(`should render correctly`, () => {
    const props = {
      match: {
        id: 2,
        url: "/",
        params: { id: "1" },
        isExact: false,
        path: "/",
      },
      history: {} as any,
      location: {} as any,
    };

    const wrapper = shallow(<CompetitionDetailsHooks {...props} />);

    expect(wrapper).toMatchSnapshot();
  });
});
