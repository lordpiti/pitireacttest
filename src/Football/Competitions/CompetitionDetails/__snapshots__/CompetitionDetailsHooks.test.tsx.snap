// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SDGDG /> with all data should render correctly 1`] = `
<div>
  <div
    className="CompetitionDetails"
  >
    <div
      className="row"
    >
      <div
        className="col-sm-3"
      >
        <WithStyles(ListItemComposition)
          itemList={
            Array [
              Object {
                "name": "Summary",
                "url": "//overview",
              },
              Object {
                "name": "Rounds",
                "url": "//competition-rounds",
              },
              Object {
                "name": "Statistics",
                "url": "//competition-statistics",
              },
            ]
          }
        >
          <div
            className="margin-bottom-medium"
          >
            <img
              className="roundedImage"
              height="50"
              src="super"
              width="50"
            />
            <span>
              undefined
            </span>
          </div>
        </WithStyles(ListItemComposition)>
      </div>
      <div
        className="col-sm-9"
      >
        <Route
          exact={true}
          path="//"
          render={[Function]}
        />
        <Route
          path="//competition-rounds/match/:id"
        />
        <Route
          path="//overview"
          render={[Function]}
        />
        <Route
          component={[Function]}
          exact={true}
          path="//competition-rounds"
        />
        <Route
          exact={true}
          path="//competition-statistics"
          render={[Function]}
        />
      </div>
    </div>
  </div>
</div>
`;
